include/BSMPT/minimum_tracer/minimum_tracer.h

```diff
@@ -0,0 +1,830 @@
+   * be global minimum
+   * @param MinTracerIn MinTracer pointer
+   */
+  Phase(double initialT,
+        std::vector<double> phase_start,
+        double LowT,
+        double HighT,
+        double &globMinEndT,
+        std::shared_ptr<MinimumTracer> &MinTracerIn);
+
+  /**
+   * @brief Construct a new Phase object
+   *
+   * @param phase_start Initial starting point for the phase
+   * @param initialT Initial temperature
+   * @param finalT Final temperature
+   * @param globMinEndT Temperature where phase is first detected to no longer
+   * be global minimum
+   * @param MinTracerIn MinTracer pointer
+   */
+  Phase(std::vector<double> phase_start,
+        double initialT,
+        double finalT,
+        std::shared_ptr<MinimumTracer> &MinTracerIn);
+
+  /**
+   * @brief Construct a new Phase:: Phase object
+   *
+   * @param phase_start Initial starting point for the phase
+   * @param initialT Temperature of the phase given as input
+   * @param LowT Lowest temperature
+   * @param HighT Highest temperature
+   * @param MinTracerIn MinTracer pointer
+   */
+  Phase(double initialT,
+        std::vector<double> phase_start,
+        double LowT,
+        double HighT,
+        std::shared_ptr<MinimumTracer> &MinTracerIn);
+
+  /**
+   * @brief Construct a new Phase:: Phase object
+   *
+   * @param initialT Temperature of the phase given as input
+   * @param LowT Lowest temperature
+   * @param HighT Highest temperature
+   * @param MinTracerIn MinTracer pointer
+   */
+  Phase(const double &initialT,
+        const double &LowT,
+        const double &HighT,
+        std::shared_ptr<MinimumTracer> &MinTracerIn);
+};
+
+/**
+ * @brief CoexPhases struct to save pair of coexisting phases (false and true
+ * phase)
+ */
+struct CoexPhases
+{
+  int coex_pair_id;
+  double T_high;
+  double T_low;
+  Phase false_phase;
+  Phase true_phase;
+  double crit_temp        = -1;
+  std::string crit_status = "not_set";
+
+  /**
+   * @brief empty constructor
+   */
+  CoexPhases();
+
+  /**
+   * @brief constructor
+   */
+  CoexPhases(const int &pair_id,
+             const Phase &false_phase,
+             const Phase &true_phase,
+             const double &Tlow_in,
+             const double &Thigh_in);
+
+  /**
+   * @brief CalculateTc critical temperature for coexising phase pair
+   */
+  void CalculateTc();
+};
+
+/**
+ * @brief Complete vacuum structure of the theory for this parameter point
+ *
+ */
+struct Vacuum
+{
+  /**
+   * @brief Lowest temperature 0 GeV
+   *
+   */
+  double T_low = 0;
+
+  /**
+   * @brief Highest temperature, 300 GeV or set in input file
+   *
+   */
+  double T_high = 300;
+
+  /**
+   * @brief Lowest temperature at which high-temperature phase is found to
+   * exist
+   *
+   */
+  double T_low_highTempPhase;
+
+  /**
+   * @brief Highest temperature at which low-temperature phase is found to
+   * exist
+   *
+   */
+  double T_high_lowTempPhase;
```

Comment:
initial values?

Reply:
They get set to values inside the `Vacuum::MultiStepPTMode`-functions. We could initialize them with e.g. `-1` or something?

Reply:
Done.